### ISSUE TITLE
Improve accessibility of inline message

### ISSUE THREAD

**Issue by anna-lach** [sl-design-system/components#1024]
According to Michael's question from Sprint review (05.03.2024) we need to improve accessibility of the inline message component.
First we need to make some research again.
The main problem is when using `'role'` in the component all the content is read by default by a Screen Reader, even a close button, which ideally should be written only when it gets focus - when user navigates to it, not by default.

![image](https://github.com/sl-design-system/components/assets/111562742/21466c88-0cd7-44b1-98f4-258a176c4108)

More details: https://sanoma.slack.com/archives/C03SA9HUUA3/p1709649841422769 

```[tasklist]
### Tasks
```

**Comment by anna-lach** [comment]
Maybe some helpful information can be found here: https://carbondesignsystem.com/components/notification/accessibility#how-it-works

<img width="800" alt="image" src="https://github.com/sl-design-system/components/assets/111562742/10f5d194-ed52-46d8-bf10-389754d410f0">

**Comment by anna-lach** [comment]
1. Try with moving role to the 'content' element - there might be a problem with Screen Readers and shadow DOM
2. Maybe we will need a kind of Announcer https://dev.to/steady5063/aria-live-in-javascript-frameworks-4f9p that people can use in their apps

We need to finish the discussion next time.

**Comment by anqaka** [comment]
I've review this component and currently I see following issues: 
1. the message is not announce to the screen reader, because live region (role="alert", role="status", aria-live attribute ) is injected into the page when the message appears. It should be added to the container before the message appears. 

2. Live region doesn’t handle rich text. so the "button" element won't be announced. only the Label of the button. 
3. I didn't find option to not announce a part of the element injected to the live region. 
4. The next important question: does notification message should have the "Close" button (or any other action)?

According to the ARIA specification: 

> Authors SHOULD ensure an element with role status does not receive focus as a result of change in status. […]
> Neither authors nor user agents are required to set or manage focus to an alert in order for it to be processed. Since alerts are not required to receive focus, authors SHOULD NOT require users to close an alert. If an author desires focus to move to a message when it is conveyed, the author SHOULD use alertdialog instead of alert. 

If we remove the "Close" button, and it disappears after for example ~500ms, will it be still accessible for users with 
cognitive limitation who need more time to access the information? 

If we move focus, how we can ensure that user go back to the previous action after closing the message? 

I feel like it requires more discussion to find the best usable solution for all users (mouse, keyboard, assistive technologies etc). 

I share here 2 great articles with complex explanation what is live region how and when to use it: 

- [Accessible notifications with ARIA Live Regions (Part 1)](https://www.sarasoueidan.com/blog/accessible-notifications-with-aria-live-regions-part-1/)
- [Accessible notifications with ARIA Live Regions (Part 2)](https://www.sarasoueidan.com/blog/accessible-notifications-with-aria-live-regions-part-2/)

**Comment by Diaan** [comment]
This was in limbo; status was "Waiting for input" but not assigned to a sprint so we didn't see it. Can we close this?

**Comment by anna-lach** [comment]
According to the accessibility slack channel discussion: https://sanoma.slack.com/archives/CFRULMYLD/p1747122952057719,
we've removed `action` slot and examples with action buttons (we do not recommend adding interactive elements inside the inline message component, interactive elements are not properly announced by aria-live). The only exception is `close` button, but the `close` button is not being announced by a screen reader there.
We've removed `role` from the inline-message and we're using `announcer` instead of it. 
For the `danger` `variant` there is an `aria-live="assertive"` used, for the rest: `aria-live="polite"`.
